### DISCUSSION TITLE
Make rustdesk work in company networks

### DISCUSSION THREAD

**Discussion by sbrkx** [rustdesk/rustdesk-server-pro#300] [Q&A]
Hi,
we just purchased a licence for rustdesk pro.
I spent the whole week trying to make rustdesk work in a very common company scenario where only standard web ports are open for outgoing connections. 
What I tried to achieve was to make the relay server listen to port :443 and the id/rendezvous server listen on port :80
Although I got everything to work on the server side where I could telnet both, it is impossible to let the clients connect with these settings. It is as if there are hardcoded ports somehow because even in the client service logs I don't find any hints that there is a connection issue.
My settings for custom client:
Host: myhostname:80
API: standard https:// url behind nginx
Relay server: relay.myhostname:443
Default settings:
RENDEZVOUS_SERVER=myhostname:80

Can someone help me with this? 

The only thing I found was a hardcoded 21116 port in the source code but it should work more easy?

Thanks!

**Comment by rustdesk** [comment]
I would suggest you use http proxy instead.

**Reply by sbrkx** [reply]
So basically I host my own http proxy on the hbbs server side? And forward port 443->21116?

**Reply by rustdesk** [reply]
Apologize, missed your reply.

I meant set the HTTP proxy on the RustDesk client side. Then, all data will go through this proxy, using port 80 (443) for standard HTTP traffic.
https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#proxy-url

80(443) <-> http(s) proxy server <-> rustdesk server pro (21116 etc)
